spark/src/main/scala/geotrellis/spark/io/hadoop/geotiff/GeoTiffLayerReader.scala

```diff
@@ -0,0 +1,118 @@
+package geotrellis.spark.io.hadoop.geotiff
+
+import geotrellis.raster.io.geotiff._
+import geotrellis.raster.{CellGrid, Raster, RasterExtent}
+import geotrellis.raster.resample.{RasterResampleMethods, ResampleMethod}
+import geotrellis.spark.tiling.ZoomedLayoutScheme
+import geotrellis.spark.{LayerId, SpatialKey}
+import geotrellis.vector.{Extent, ProjectedExtent}
+import geotrellis.raster.crop.Crop
+import geotrellis.raster.reproject.Reproject.{Options => ReprojectOptions}
+import geotrellis.raster.io.geotiff.reader.GeoTiffReader
+import geotrellis.raster.reproject.RasterReprojectMethods
+import geotrellis.raster.merge.RasterMergeMethods
+import geotrellis.util.ByteReader
+
+import scalaz.concurrent.{Strategy, Task}
+import scalaz.stream.{Process, nondeterminism}
+
+import java.net.URI
+import java.util.concurrent.{ExecutorService, Executors}
+
+import scala.reflect.ClassTag
+
+/** Approach with TiffTags stored in a DB */
+trait GeoTiffLayerReader[M[T] <: Traversable[T]] {
```

Comment:
Confused between the `geotiff` and `cog` packages. Is the `geotiff` stuff the unstructured layers?

Reply:
Yes, `geotiff` means unstructured. Do you think it would be better to put it into `cog` package and to rename everything or to rename this `geotiff` package into `ucog` or smth like that?